vm/src/frame.rs

```diff
@@ -995,19 +1013,27 @@ impl Frame {
         Ok(None)
     }
 
-    fn load_name(&self, vm: &VirtualMachine, name: &str) -> FrameResult {
-        match self.scope.load_name(&vm, name) {
-            Some(value) => {
-                self.push_value(value);
-                Ok(None)
-            }
-            None => {
-                let name_error_type = vm.ctx.exceptions.name_error.clone();
-                let msg = format!("name '{}' is not defined", name);
-                let name_error = vm.new_exception(name_error_type, msg);
-                Err(name_error)
-            }
-        }
+    fn load_name(
+        &self,
+        vm: &VirtualMachine,
+        name: &str,
+        scope: &bytecode::NameScope,
+    ) -> FrameResult {
+        let value = match scope {
+            bytecode::NameScope::Global => self.scope.globals.get_item(name, vm)?,
+            bytecode::NameScope::Local => match self.scope.load_name(&vm, name) {
+                Some(value) => value,
+                None => {
+                    let name_error_type = vm.ctx.exceptions.name_error.clone();
```

Comment:
The match/exception creation here should be done outside the ```match scope```. As it is I think load_global will raise ```KeyError```rather than ```NameError```.

Reply:
Agreed, moved it out of the match.